### PR TITLE
Add support for ITIMER_REAL

### PR DESCRIPTION
**Pull request by jonashaag**

### CODE REVIEW

**Comment by fijal** [comment]
At the very least this needs an extensive section in documentation/docstring, not just a comment in source code

**Comment by jonashaag** [comment]
Agreed, I wanted to check with you first if the implementation is OK

**Comment by fijal** [comment]
Note that it also has some unrelated changes there?

File: src/vmprof_main.h [hunk 1]
```diff
@@ -95,7 +95,7 @@ static int proc_file = -1;
 
 static int get_stack_trace(void** result, int max_depth, ucontext_t *ucontext)
 {
-    PyThreadState* current = get_current_thread_state();
+    PyThreadState *current = get_current_thread_state();
```
Comment (jonashaag) [inline src/vmprof_main.h]: This should be the only unrelated change

File: src/vmprof_common.h [hunk 2]
```diff
@@ -108,9 +114,52 @@ static int opened_profile(char *interp_name, int memory)
...
+    PyThreadState *tstate;
 #if PY_MAJOR_VERSION >= 3
-    return (PyThreadState*)_Py_atomic_load_relaxed(&_PyThreadState_Current);
+    tstate = (PyThreadState*)_Py_atomic_load_relaxed(&_PyThreadState_Current);
 #else
-    return _PyThreadState_Current;
+    tstate = _PyThreadState_Current;
 #endif
+
+    if (tstate == NULL) {
```
Comment (fijal) [inline src/vmprof_common.h]: This change is massive, unrelated to the pull request and potentially problematic. The problem here is that we don't hold any lock while walking the interpreter state - that means that if we have more than one thread, it will likely explode. Note that cpython has a lock to do such manipulations, but it's simply not exposed.

Comment (jonashaag) [inline src/vmprof_common.h]: It isn't unrelated. The problem is that `tstate` is always `NULL` when waiting for a syscall, so if we didn't look for the thread in the interpreter state, we'd never get any profiling samples for the program while it's waiting for a syscall. It wouldn't make sense to use `ITIMER_REAL` without this code.

Comment (fijal) [inline src/vmprof_common.h]: I understand, I've tried using good APIs from CPython, unfortunately this is not safe (and will end up in crashes), since you can't walk this list without holding a lock - it's not a given that some other thread is not running at the same time (even if your thread is suspended right now)

**Comment by jonashaag** [comment]
Since there doesn't seem to be a safe way to solve the interpreter state problem I'll stop working on this.

For future reference, if anyone wants to continue working on it: This needs a change in CPython (haven't looked into PyPy so far).
